File: scripts/random-image.js
const fs = require("fs");
const path = require("path");

const FILE_TO_UPDATE = path.resolve(__dirname, "../README.md");
const TAENGOO_MAX = 9;
const WINTER_MAX = 9;
const data = fs
    .readFileSync(FILE_TO_UPDATE, "utf-8")
    .replace(
        /images\/taengoo[0-9]+/,
        `images/taengoo${Math.floor(Math.random() * TAENGOO_MAX) + 1}`
    )
    .replace(
        /images\/winter[0-9]+/,
        `images/winter${Math.floor(Math.random() * WINTER_MAX) + 1}`
    );

fs.writeFileSync(FILE_TO_UPDATE, data, "utf-8");

File: scripts/random-image.cpp
#include <time.h>

#include <iostream>
#include <fstream>
#include <random>

#define FILE_TO_REPLACE "README.md"
#define FILE_TO_WRITE "README.bak.md"
#define TAENGOO_MAX 18
#define WINTER_MAX 23

using namespace std;

int main()
{
    random_device rd;
    mt19937 gen(rd());
    string filePath = FILE_TO_REPLACE;
    string tmpPath = FILE_TO_WRITE;
    ifstream openFile(filePath.data());
    ofstream writeFile(tmpPath.data());
    uniform_int_distribution<int> taeng_random(1, TAENGOO_MAX);
    uniform_int_distribution<int> winter_random(1, WINTER_MAX);

    if (!openFile || !writeFile)
    {
        printf("Can't open files\n");
        return 1;
    }

    string line;
    while (getline(openFile, line))
    {
        if (line.find("alt=\"탱구\"", 0) != string::npos)
        {
            line.reserve(256);
            line = "<img src=\"https://marshallku.github.io/marshallku/assets/images/taengoo" + to_string(taeng_random(gen)) + ".gif\" alt=\"탱구\" height=\"150\" /><img src=\"https://marshallku.github.io/marshallku/assets/images/winter" + to_string(winter_random(gen)) + ".gif\" alt=\"윈터\" height=\"150\" />";
        }

        writeFile << line << "\n";
    }

    writeFile.close();
    openFile.close();

    remove(FILE_TO_REPLACE);
    rename(FILE_TO_WRITE, FILE_TO_REPLACE);

    return 0;
}
